cluster-app-values-schema/README.md

```diff
@@ -0,0 +1,373 @@
+
+In the case of a generated user interface, only the non-deprecated subschema will be applied. All other subschemas will be ignored.
+
+Example:
+
+```json
+"replicas": {
+  "anyOf": [
+    {
+      "type": "string",
+      "deprecated": true,
+      "$comment": "to be removed in the next major version, please use the integer type instead"
+    },
+    {
+      "type": "integer",
+      "maximum": 100,
+      "title": "Size"
+    }
+  ]
+}
+```
+
+## R11: Use `deprecated` to phase out properties {#r11}
+
+To indicate that a property is supposed to be removed in a future version, the property SHOULD carry the `deprecated` key with the value `true`.
+
+As a consequence, user interfaces for cluster creation SHALL NOT display the according form field for data entry. However, providing the property within values YAML will not cause any failure.
+
+In addition, it is RECOMMENDED to add a `$comment` key to the property, with information regarding
+
+- which property will replace the deprecated one (if any)
+- when the property will be removed
+
+Note that `$comment` content is not intended for display in any UI nor processing in any tool. It is mainly targeting schema developers and anyone using the schema itself as a source of information.
+
+## R12: Provide valid string values where possible {#r12}
+
+For string properties, in some cases only a few values are considered valid. In this case, the schema SHOULD specify these selectable values in one of the following forms:
+
+1. Using `enum` in case the selectable value does not require a more user-friendly label.
+2. Using `oneOf` with a combination of `title` and `const`, in case a user-friendly label is needed in addition to the value. See below for more details.
+
+As an example of the `enum` method, we use a string property with two possible values. The values `active` and `inactive` are considered to be speaking for themselves.
+
+```json
+"autoRefresh": {
+  "type": "string",
+  "title": "Auto refresh",
+  "enum": ["active", "inactive"]
+}
+```
+
+However, if the string values are not considered self-explanatory, the `oneOf` method can be used to provide user-friendly labels in addition to the axctual value. Here, `oneOf` is an array of tuples, in which `title` contains the user-friendly description and `const` provides the value.
+
+```json
+"imagePullPolicy": {
+  "type": "string",
+  "title": "Pull policy",
+  "description": "Whether the container image should be pulled from a registry",
+  "oneOf": [
+    {"const": "IfNotPresent", "title": "Pull image if not present in the node"},
+    {"const": "Always", "title": "Always pull up-to-date image"},
+    {"const": "Never", "title": "Never pull image, use image available locally"}
+  ]
+}
+```
+
+### R13: Avoid recursion {#r13}
+
+The JSON Schema keywords `dynamicRef`, `dynamicAnchor`, and `recursiveRef` MUST NOT be used.
```

Comment:
The title says to avoid recursion. Do we forbid recursion in general or just these keywords?
I know that we talked about these keywords, because they make implementation in schemalint more cumbersome. But I guess for happa self-referencing recursion is impossible to display properly in the UI.